kernel/ixgbe/src/lib.rs

```diff
@@ -1213,29 +1302,29 @@ pub enum FilterProtocol {
 }
 
 /// A helper function to poll the nic receive queues (only for testing purposes).
-fn rx_poll_mq(qid: usize) -> Result<ReceivedFrame, &'static str> {
-    let nic_ref = get_ixgbe_nic().ok_or("ixgbe nic not initialized")?;
+fn rx_poll_mq(qid: usize, nic_id: PciLocation) -> Result<ReceivedFrame, &'static str> {
+    let nic_ref = get_ixgbe_nic(nic_id).ok_or("ixgbe nic not initialized")?;
     let mut nic = nic_ref.lock();      
     nic.rx_queues[qid as usize].poll_queue_and_store_received_packets()?;
     let frame = nic.rx_queues[qid as usize].return_frame().ok_or("no frame")?;
     Ok(frame)
 }
 
 /// A helper function to send a test packet on a nic transmit queue (only for testing purposes).
-fn tx_send_mq(qid: usize) -> Result<(), &'static str> {
+fn tx_send_mq(qid: usize, nic_id: PciLocation) -> Result<(), &'static str> {
     let packet = test_packets::create_dhcp_test_packet()?;
-    let nic_ref = get_ixgbe_nic().ok_or("ixgbe nic not initialized")?;
+    let nic_ref = get_ixgbe_nic(nic_id).ok_or("ixgbe nic not initialized")?;
```

Comment:
should this error message be changed to reflect that the requested nic_id didn't exist (or wasn't inited)?

Reply:
same as above